### PR TITLE
add multi-tenant feature

### PR DESCRIPTION
**Pull request by xxwjj**
**What this PR does / why we need it**:
Add multi-tenant feature

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
Fxied #294 
if you use cli,  some EVNs need to be mentioned:

1. OPENSDS_AUTH_STRATEGY: The authentication strategy, the value can be keystone or noauth, default is  noauth.
2. OPENSDS_TENANT_ID: If you authentication strategy that you specified is onauth, you can use this ENV that set different uuid to test multi-tenant.
3. if the strategy is keystone, the following EVNs need to be setted.The usage is the same as openstack cli.
OS_AUTH_URL
OS_USERNAME
OS_PASSWORD
OS_TENANT_NAME
OS_PROJECT_NAME
OS_USER_DOMAIN_ID

**Special notes for your reviewer**:
@leonwanghui @xiangrumei @xing-yang  @wisererik 

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/model/pool.go [hunk 1]
```diff
@@ -24,6 +24,10 @@ package model
 // atomic pool and can be abstracted from any storage platform.
 type StoragePoolSpec struct {
 	*BaseModel
+	// The uuid of project
+	// + readOnly
+	ProjectId string `json:"projectId"`
```
Comment (leonwanghui) [inline pkg/model/pool.go]: Should be changed to `TenantId`

File: pkg/model/volume.go [hunk 2]
```diff
@@ -69,6 +69,7 @@ type VolumeSpec struct {
 // VolumeAttachmentSpec is a description of volume attached resource.
 type VolumeAttachmentSpec struct {
 	*BaseModel
+
 	// The uuid of the project that the volume belongs to.
```
Comment (leonwanghui) [inline pkg/model/volume.go]: `TenantId`

File: client/receiver.go [hunk 3]
```diff
@@ -16,39 +16,70 @@ package client
 
 import (
 	"encoding/json"
-	"errors"
 	"fmt"
+	"github.com/astaxie/beego/httplib"
```
Comment (leonwanghui) [inline client/receiver.go]: Please use `go fmt` to update the code style.

File: pkg/model/pool.go (hunk 1 above)
Comment (xxwjj) [reply pkg/model/pool.go]: I will commit a another PR to solve this issue.

**Comment by xxwjj** [comment]
@leonwanghui  The usage of client, you can find an example in osdslet/cli/cli.go.  You just need to fill up the paramters in KeystonAuthOptions structrue, and assign this option to Config structrue. then you can new a opensds client.

**Pull request merged**